src/puya/compile.py

```diff
@@ -172,50 +170,61 @@ def module_irs_to_teal(
     return result
 
 
-def _get_python_exe() -> str | None:
+def _get_python_executable() -> str | None:
+    prefix = _get_prefix()
+    if not prefix:
+        logger.warning("Could not determine python prefix or algopy version")
+        return None
+    logger.info(f"Found python prefix: {prefix}")
+    venv_paths = sysconfig.get_paths(vars={"base": prefix})
+
+    # use sys.executable as cross-platform way of correctly using python.exe or python
+    python_exe = Path(venv_paths["scripts"]) / Path(sys.executable).name
```

Comment:
There's no guarantee that the `sys.executable` name will match that of the target environment